common/server_player.cpp

```diff
@@ -1069,9 +1069,15 @@ Server_Player::cmdRollDie(const Command_RollDie &cmd, ResponseContainer & /*rc*/
         return Response::RespContextError;
     }
 
+    const auto validatedSides = static_cast<int>(std::min(std::max(cmd.sides(), MINIMUM_DIE_SIDES), MAXIMUM_DIE_SIDES));
+    const auto validatedDiceToRoll =
+        static_cast<int>(std::min(std::max(cmd.dice_to_roll(), MINIMUM_DICE_TO_ROLL), MAXIMUM_DICE_TO_ROLL));
+
     Event_RollDie event;
-    event.set_sides(cmd.sides());
-    event.set_value(rng->rand(1, cmd.sides()));
+    event.set_sides(validatedSides);
+    for (auto i = 0; i < validatedDiceToRoll; ++i) {
```

Comment:
would it be worth optimising this? one of these rand calls throws away a lot of the "entropy" for one roll, multiple dice could be randomised using just one rand call and a %

Reply:
Agreed, will do

Reply:
did you change anything here?

Reply:
I looked to change it up, and it made it overly complex. I'm not sure what you're looking for on this